### PR TITLE
Fix: Fixed issue with appending "copy" instead of "#"

### LINKED ISSUE #11697: Bug: Generating new name for conflict is appending "copy" instead of a number "(#)"
### What feature or improvement do you think would benefit Files?

When I move or copy a file from one folder to another folder that already contains a file with the same name, and I would like to keep both, I would like the option for it to append " (#)" to the file (like Windows) instead of "- Copy" when I select Generate New Name. 

If it were the same file, I would almost always choose Replace or Skip, not Keep Both (aka Generate New Name), so when I choose to keep both, it is almost always because it isn't the same file, and therefore is not a copy, but an additional item with the same name.

### Requirements

- [ ] When copying and pasting items within the same folder, "- copy" should be appended (as is File Explorer)
- [ ] When copying and pasting item from a different folder, selecting generate new name from the conflict modal should append a number (as in File Explorer)

### Files Version

2.4.40.0

### Windows Version

Windows 11 22H2 22623.1020

### Comments

_No response_

### PR DESCRIPTION
**Pull request by Mohammed-619**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #11697

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [x] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?
   1. Open app ...
   2. Click settings button ...

**Screenshots (optional)**

### CODE REVIEW

**Comment by Josh65-2201** [comment]
Thanks for contributing to Files, for future pull request
- In the title of the issue put `Feature:` or `Fix:` depending on what the PR is about. In this case it would be `Fix:`
- Please add the issue number were it says `Closes #issue...` it should look like this `Closes #11697`
- In the square brackets where you put yes replace the space with a `x` it should look like `[x]` and on the preview tab it should show a check mark.

File: src/Files.App/Utils/Storage/Helpers/FileOperationsHelpers.cs [hunk 1]
```diff
@@ -439,6 +440,34 @@ public static Task<(bool, ShellOperationResult)> MoveItemAsync(string[] fileToMo
...
+			if (destination == null)
+			{
+				throw new InvalidOperationException("Destination folder is not accessible.");
+			}
+
+			string newName = originalName;
+			int counter = 2;
+
+			// Iterate through the children of the destination folder and check if the new name already exists.
+			// If it does, append a counter to make it unique.
```
Comment (QuaintMako) [inline src/Files.App/Utils/Storage/Helpers/FileOperationsHelpers.cs:459]: The for loop speaks for itself, no need to document its functionality.

File: src/Files.App/Utils/Storage/Helpers/FileOperationsHelpers.cs [hunk 2]
```diff
@@ -439,6 +440,34 @@ public static Task<(bool, ShellOperationResult)> MoveItemAsync(string[] fileToMo
...
+			foreach (var existingItem in destination.EnumerateChildren(FolderItemFilter.Storage))
+			{
+				if (string.Equals(newName, existingItem.Name, StringComparison.OrdinalIgnoreCase))
+				{
+					newName = $"{originalName} ({counter})";
```
Comment (QuaintMako) [inline src/Files.App/Utils/Storage/Helpers/FileOperationsHelpers.cs:464]: I'd suggest building the newname only when the counter value is calculated. It's will prevent unnecessary string assignation. In the case of a lot of files sharing the same name, it'd reduce the cost by quite a margin.

File: src/Files.App/Utils/Storage/Helpers/FileOperationsHelpers.cs [hunk 3]
```diff
@@ -439,6 +440,34 @@ public static Task<(bool, ShellOperationResult)> MoveItemAsync(string[] fileToMo
...
+					counter++;
+				}
+			}
+
+			return newName;
```
Comment (QuaintMako) [inline src/Files.App/Utils/Storage/Helpers/FileOperationsHelpers.cs:469]: Could return directly `return $"{originalName} ({counter})";`

File: src/Files.App/Utils/Storage/Helpers/FileOperationsHelpers.cs (hunk 2 above)
Comment (QuaintMako) [reply src/Files.App/Utils/Storage/Helpers/FileOperationsHelpers.cs:464]: In the end, this foreach can be compacted into a linq request that returns the number of elements with name `originalName` inside the folder.
Might be worth to do the change for the readability gain.

**Review by QuaintMako — COMMENTED** [review]
I cannot test the functionality, but a few feedback on the existing code already.

**Comment by yaira2** [comment]
@Mohammed-619 thank you for the PR. 

Can you take a moment to fill out the section for "Are there any other steps that were used to validate these changes" so we can follow the step on our side?

**Comment by Mohammed-619** [comment]
Move a file/folder into a folder that has file/folder with the same name.
for example move /Desktop/filename into /Desktop/folder, but /Desktop/folder has to have an item name filename. Please let me know if it didn’t work or if this different from what expected.

Sent from my iPhone

On 19 Oct 2023, at 5:43 PM, Yair ***@***.***> wrote:

﻿

[EXTERNAL EMAIL]

@Mohammed-619<https://urldefense.com/v3/__https://github.com/Mohammed-619__;!!JmPEgBY0HMszNaDT!oA-QZiO1_6k2Zd9A1VpZzuAp0n7rkSdRJOSMv1OUH64xSss8jJA0IZZfyWxoH0zEJPfiWj5kw9QSxS1g_QTq6xDktAqRNy8$> thank you for the PR.

Can you take a moment to fill out the section for "Are there any other steps that were used to validate these changes" so we can follow the step on our side?

**Comment by yaira2** [comment]
@Mohammed-619 I'll test later but can you confirm that the conflicts dialog is still displayed?

**Pull request closed without merging**